cg/meta/workflow/taxprofiler.py

```diff
@@ -101,24 +112,73 @@ def write_sample_sheet(
+            file_path=NextflowAnalysisAPI.get_params_file_path(
+                case_id=case_id, root_dir=self.root_dir
+            ),
+        )
+
+    def get_default_parameters(self, case_id: str) -> Dict:
+        """Returns a dictionary with default Taxprofiler parameters."""
+        return {
+            "input": NextflowAnalysisAPI.get_input_path(
+                case_id=case_id, root_dir=self.root_dir
+            ).as_posix(),
+            "databases": self.databases,
+            "outdir": NextflowAnalysisAPI.get_outdir_path(
+                case_id=case_id, root_dir=self.root_dir
+            ).as_posix(),
+            "save_preprocessed_reads": TaxprofilerDefaults.SAVE_PREPROCESSED_READS,
+            "perform_shortread_qc": TaxprofilerDefaults.PERFORM_SHORTREAD_QC,
+            "perform_shortread_complexityfilter": TaxprofilerDefaults.PERFORM_SHORTREAD_COMPLEXITYFILTER,
+            "save_complexityfiltered_reads": TaxprofilerDefaults.SAVE_COMPLEXITYFILTERED_READS,
+            "perform_shortread_hostremoval": TaxprofilerDefaults.PERFORM_SHORTREAD_HOSTREMOVAL,
+            "hostremoval_reference": self.hostremoval_reference,
+            "save_hostremoval_index": TaxprofilerDefaults.SAVE_HOSTREMOVAL_INDEX,
+            "save_hostremoval_mapped": TaxprofilerDefaults.SAVE_HOSTREMOVAL_MAPPED,
+            "save_hostremoval_unmapped": TaxprofilerDefaults.SAVE_HOSTREMOVAL_UNMAPPED,
+            "perform_runmerging": TaxprofilerDefaults.PERFORM_RUNMERGING,
+            "run_kraken2": TaxprofilerDefaults.RUN_KRAKEN2,
+            "kraken2_save_reads": TaxprofilerDefaults.KRAKEN2_SAVE_READS,
+            "kraken2_save_readclassification": TaxprofilerDefaults.KRAKEN2_SAVE_READCLASSIFICATION,
+            "run_krona": TaxprofilerDefaults.RUN_KRONA,
+            "run_profile_standardisation": TaxprofilerDefaults.RUN_PROFILE_STANDARDISATION,
+            "priority": self.account,
+            "clusterOptions": f"--qos={self.get_slurm_qos_for_case(case_id=case_id)}",
+        }
+
     def config_case(
         self,
         case_id: str,
         instrument_platform: SequencingPlatform.ILLUMINA,
-        fasta: Optional[str],
+        dry_run: bool,
+        fasta: str = "",
     ) -> None:
         """Create sample sheet file for Taxprofiler analysis."""
         self.create_case_directory(case_id=case_id)
         LOG.info("Generating sample sheet")
         self.write_sample_sheet(
-            case_id=case_id,
-            instrument_platform=instrument_platform,
-            fasta=fasta,
+            case_id=case_id, instrument_platform=instrument_platform, fasta=fasta, dry_run=dry_run
         )
         LOG.info("Sample sheet written")
+        LOG.info("Generating parameters file")
+        self.write_params_file(case_id=case_id, dry_run=dry_run)
+        if dry_run:
+            LOG.info("Dry run: Config files will not be written")
+            return
+
+        LOG.info("Configs files written")
```

Comment:
Could you review the Logs here and relocate those that belong specifically to functions into their corresponding methods? This way, if we eventually re-use `write_params_file` for another task, we won't have to insert these logs twice 👍

Reply:
Done.